### PR TITLE
Order runs by `started_at` date

### PR DESCRIPTION
**Pull request by phixMe**
This changes the API to return runs ordered by the `STARTED_AT` (`transitioned_at` in `run_states` table) time instead of the `created_at` time.

### CODE REVIEW

File: api/src/main/java/marquez/db/RunDao.java
Comment (wslulciuc) [inline api/src/main/java/marquez/db/RunDao.java:154]: I think this should be:

```
ORDER BY r.started_at DESC NULLS LAST
```

**Comment by wslulciuc** [comment]
@phixMe: Mind also adding / updating an integration test for listing runs to [`MarquezAppIntegrationTest`](https://github.com/MarquezProject/marquez/blob/main/api/src/test/java/marquez/MarquezAppIntegrationTest.java)?

File: api/src/main/java/marquez/db/RunDao.java
Comment (phixMe) [reply api/src/main/java/marquez/db/RunDao.java:154]: The runs table doesn't have a `started_at` time. This is using the alias from the `SELECT_RUN` constant which is the joined in `transitioned_at` column from `run_states`.

Comment (wslulciuc) [reply api/src/main/java/marquez/db/RunDao.java:154]: Ahh, I see 👍

File: api/src/main/resources/marquez/db/migration/V23__index_transitioned_at_on_run_states.sql
Comment (wslulciuc) [inline api/src/main/resources/marquez/db/migration/V23__index_transitioned_at_on_run_states.sql:2]: Adding this index seems fine with me, @henneberger thoughts?

**Review by wslulciuc — APPROVED** [review]
LGTM 🥇

File: api/src/main/resources/marquez/db/migration/V23__index_transitioned_at_on_run_states.sql
Comment (henneberger) [reply api/src/main/resources/marquez/db/migration/V23__index_transitioned_at_on_run_states.sql:2]: Should this on be `runs`, not `run_states`?

Comment (henneberger) [reply api/src/main/resources/marquez/db/migration/V23__index_transitioned_at_on_run_states.sql:2]: Oh, that is in the PR that I've been trying to get landed.

Comment (wslulciuc) [reply api/src/main/resources/marquez/db/migration/V23__index_transitioned_at_on_run_states.sql:2]: Yeah, currently, the `transitioned_at` column is only on the `run_states` table.

Comment (henneberger) [reply api/src/main/resources/marquez/db/migration/V23__index_transitioned_at_on_run_states.sql:2]: This index isn't good since we aren't trying to order the entire run state table by transition time.  Can we wait until that gets landed and have an index that does:
```
create index run_transitioned_at_index
    on run_states (namespace_name, job_name, transitioned_at desc);
    ```

**Comment by wslulciuc** [comment]
FYI, we can still merge the PR though the integration test for Airflow is failing (it's unrelated and currently fails only for forks)

File: api/src/main/resources/marquez/db/migration/V23__index_transitioned_at_on_run_states.sql
Comment (wslulciuc) [reply api/src/main/resources/marquez/db/migration/V23__index_transitioned_at_on_run_states.sql:2]: Sure, so remove the `transitioned_at` index for now? /cc @phixMe

Comment (phixMe) [reply api/src/main/resources/marquez/db/migration/V23__index_transitioned_at_on_run_states.sql]: @wslulciuc @henneberger Sounds good, removing it.

**Pull request merged**
